src/types/plugin-input.ts

```diff
@@ -24,10 +28,33 @@ export const startStopSchema = T.Object({
   ),
   miscellaneous: T.Object(
     {
-      maxConcurrentTasks: T.Number(),
       startRequiresWallet: T.Boolean(),
+      maxConcurrentTasks: T.Array(
+        T.Object({
+          role: userRoleSchema,
```

Comment:
What happens if I add a role into my config like "newbie" or "gilded" and my org has it too, would this produce a runtime error because `userRoleSchema` only supports the three you've listed?

[this](https://docs.github.com/en/enterprise-cloud@latest/admin/overview/setting-up-a-trial-of-github-enterprise-cloud) trial would allow proper QA as it allows you to create custom roles as an enterprise otherwise unit tests would suffice

[This endpoint](https://docs.github.com/en/rest/orgs/organization-roles?apiVersion=2022-11-28#get-all-organization-roles-for-an-organization) returns the roles that an organization has and the response schema shows how varied it might be. Consider this, and extend the typebox schemas so that they are fully extendable